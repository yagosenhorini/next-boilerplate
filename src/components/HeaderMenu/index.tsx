'use client';

import Link from 'next/link';
import cx from 'classnames';

import { headerMenu } from '@Resources/header-menu';

import * as S from './styled';

export default function HeaderMenu() {
  return (
    <>
      <S.Menu data-testid="header-menu">
        <S.LogoWrapper>
          <S.Image
            width={40}
            height={40}
            src="/img/images/logo-teste.png"
            alt="Logo"
            title="Logo"
          />
        </S.LogoWrapper>
        <S.HeaderMenuWrapper>
          {headerMenu.map(({ route, title, main }) => (
            <Link key={route} href={route}>
              <S.HeaderMenuItem id={route} className={cx({ 'is--home': main })}>
                {title}
              </S.HeaderMenuItem>
            </Link>
          ))}
        </S.HeaderMenuWrapper>
      </S.Menu>
    </>
  );
}
