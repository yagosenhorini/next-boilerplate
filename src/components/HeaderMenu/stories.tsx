import { Meta, StoryObj } from '@storybook/react';
import HeaderMenu from '.';

export default {
  title: 'HeaderMenu',
  component: HeaderMenu,
  tags: ['autodocs'],
} as Meta;

export const Default: StoryObj = {};
