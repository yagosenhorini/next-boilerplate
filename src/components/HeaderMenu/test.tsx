import { render, screen } from '@testing-library/react';

import HeaderMenu from '.';

import StyledProvider from '../../lib/theme';

describe('<HeaderMenu />', () => {
  it('should render the component', () => {
    const { container } = render(
      <StyledProvider>
        <HeaderMenu />
      </StyledProvider>
    );

    expect(screen.getByTestId('header-menu')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
