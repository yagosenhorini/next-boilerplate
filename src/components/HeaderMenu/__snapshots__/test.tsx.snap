// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderMenu /> should render the component 1`] = `
.c0 {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c1 {
  cursor: pointer;
  width: 2.5rem;
  height: 2.5rem;
  margin-top: 0.9375rem;
  margin-left: 1.25rem;
}

.c2 {
  display: flex;
  list-style: none;
  width: 22.5rem;
  align-items: center;
  justify-content: space-around;
}

.c2 a {
  text-decoration: none;
}

.c3 {
  cursor: pointer;
  width: 3.75rem;
  color: #fff;
}

.c3.is--home {
  color: #ED7D31;
}

@media (min-width: 768px) {
  .c3 {
    color: #000;
  }
}

<nav
  class="c0"
  data-testid="header-menu"
>
  <div
    class="c1"
  >
    <img
      alt="Logo"
      class=""
      height="40"
      src="/img/images/logo-teste.png"
      title="Logo"
      width="40"
    />
  </div>
  <ul
    class="c2"
  >
    <a
      href="/"
    >
      <li
        class="c3 is--home"
        id="/"
      >
        Home
      </li>
    </a>
    <a
      href="#about"
    >
      <li
        class="c3"
        id="#about"
      >
        Sobre
      </li>
    </a>
    <a
      href="#projects"
    >
      <li
        class="c3"
        id="#projects"
      >
        Projetos
      </li>
    </a>
    <a
      href="/calculadora"
    >
      <li
        class="c3"
        id="/calculadora"
      >
        Calculadora TMB
      </li>
    </a>
  </ul>
</nav>
`;
